server/src/main/java/jass/server/util/GameUtil.java

```diff
@@ -0,0 +1,160 @@
+package jass.server.util;
+
+import jass.lib.GameMode;
+import jass.lib.message.BroadcastGameModeData;
+import jass.lib.message.ChooseGameModeData;
+import jass.lib.message.ChosenGameModeData;
+import jass.lib.message.GameFoundData;
+import jass.lib.servicelocator.ServiceLocator;
+import jass.server.entity.DeckEntity;
+import jass.server.entity.GameEntity;
+import jass.server.entity.RoundEntity;
+import jass.server.entity.TeamEntity;
+import jass.server.entity.UserEntity;
+import jass.server.eventlistener.ChosenGameModeEventListener;
+import jass.server.message.BroadcastGameMode;
+import jass.server.message.ChooseGameMode;
+import jass.server.message.GameFound;
+import jass.server.message.Message;
+import jass.server.repository.GameRepository;
+import jass.server.repository.RoundRepository;
+import jass.server.repository.TeamRepository;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+
+/**
+ * @author Thomas Weber, Manuele Vaccari
+ * @version %I%, %G%
+ * @since 0.0.1
+ */
+public final class GameUtil implements ChosenGameModeEventListener {
+    /**
+     * The logger to print to console and save in a .log file.
+     */
+    private static final Logger logger = LogManager.getLogger(GameUtil.class);
+
+    /**
+     * The client for player one.
+     */
+    private final ClientUtil clientPlayerOne;
+
+    /**
+     * The client for player two.
+     */
+    private final ClientUtil clientPlayerTwo;
+
+    /**
+     * The client for player three.
+     */
+    private final ClientUtil clientPlayerThree;
+
+    /**
+     * The client for player four.
+     */
+    private final ClientUtil clientPlayerFour;
+
+    /**
+     * The game.
+     */
+    private final GameEntity game;
+
+    /**
+     * @param clientPlayerOne   Player one.
+     * @param clientPlayerTwo   Player two.
+     * @param clientPlayerThree Player three.
+     * @param clientPlayerFour  Player four.
+     */
+    public GameUtil(final ClientUtil clientPlayerOne, final ClientUtil clientPlayerTwo, final ClientUtil clientPlayerThree, final ClientUtil clientPlayerFour) {
+        this.clientPlayerOne = clientPlayerOne;
+        UserEntity playerOne = clientPlayerOne.getUser();
+        this.clientPlayerTwo = clientPlayerTwo;
+        UserEntity playerTwo = clientPlayerTwo.getUser();
+        this.clientPlayerThree = clientPlayerThree;
+        UserEntity playerThree = clientPlayerThree.getUser();
+        this.clientPlayerFour = clientPlayerFour;
+        UserEntity playerFour = clientPlayerFour.getUser();
+
+        // Add event listeners
+        clientPlayerOne.addChosenGameModeEventListener(this);
```

Comment:
in the first round, player one gets to pick yes. But.. we don't decide based on the fact that this is player one, but instead should look at the round table and which user id gets to pick for this round.